src/math_tools.rs

```diff
@@ -46,10 +85,69 @@ mod tests {
 
     use super::*;
 
-    #[test]
-    fn test_shannon_entropy() {
-        let input = b"000111"; // 50% entropy distribution ~ 1 bit information
+    mod shannon {
+        use super::*;
+
+        #[test]
+        fn test_shannon_entropy() {
+            let input = b"000111"; // 50% entropy distribution ~ 1 bit information
+
+            assert_relative_eq!(shannon_entropy(input), 1.0);
+        }
+    }
+
+    mod chi_square {
+        use super::*;
+        use rand_core::{OsRng, RngCore};
 
-        assert_relative_eq!(shannon_entropy(input), 1.0);
+        #[test]
+        fn test_non_uniform_distribution() {
+            let uniform_distribution = [0u8; 4096];
+            let chi_square_value = chi_square_probability(&uniform_distribution);
+
+            // Expect chi-square to be relatively low for uniform random data
+            assert_eq!(
```

Comment:
I don't recall, if I needed to use `assert_relative_eq!` originally. I was afraid of comparison issues between floats. It may be a good idea to is it here as well, or drop it from `shannon_entropy` if it is not needed.

Reply:
`assert_eq` is sufficient it seems. dropped it in a separate commit (c4a11cf).